crates/interpreter/src/side_table.rs

```diff
@@ -20,45 +20,38 @@ use crate::module::Parser;
 
 pub struct SideTableView<'m> {
     pub func_idx: usize,
-    pub indices: &'m [u16], // including 0 and the length of metadata_array
-    pub metadata: &'m [u16],
+    pub indices: &'m [u8], // including 0 and the length of metadata_array
+    pub metadata: &'m [u8],
     pub branch_table_view: Metadata<'m>,
 }
 
 impl<'m> SideTableView<'m> {
-    pub fn new(parser: &mut crate::valid::Parser<'m>) -> Result<Self, Error> {
+    pub fn new(binary: &'m [u8]) -> Result<Self, Error> {
+        let num_functions =
+            bytemuck::pod_read_unaligned::<u16>(bytemuck::cast_slice(&binary[0 .. 2])) as usize;
```

Comment:
We don't need this anymore since `[u8]` is always aligned. We can simply `u16::from_le_bytes(binary[.. 2].try_into().unwrap())` (maybe with some `&` and parentheses).